### PR TITLE
 Remove False warnings from Settings engine for valid parameters

### LINKED ISSUE #7074: Wrong Warnings from Settings Engine for Valid Parameters in Configuration Framework
Issue Description:

Upon migrating to the configuration engine, the settings engine generates numerous warnings, such as:

```
2023-06-12 13:17:04.860 14       WARN  STARTUP          <main> Settings: WARNING: event.preconsensus.enableStorage is not a valid setting name.
bad name of setting in settings.txt line 25: event.preconsensus.enableStorage, true
WARNING: reconnect.active is not a valid setting name.
```

Contrary to the indications, these are erroneous warnings. The flagged parameters are valid within the context of the new configuration framework, yet they prompt warnings as they are not identified by the legacy settings engine.

While this issue does not result in any immediate functional problems, it introduces a potential risk. This misleading warning system may inadvertently condition users to disregard these types of alerts, which could lead to overlooking actual configuration issues in the future.

### PR DESCRIPTION
**Pull request by timo0**
Fix #7074

### CODE REVIEW

**Review by hendrikebbers — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SettingConstants.java [hunk 1]
```diff
@@ -101,5 +103,97 @@ public final class SettingConstants {
     static final String PLAYBACK_END_TIME_STAMP_DEFAULT_VALUE = "";
     static final boolean GOSSIP_WITH_DIFFERENT_VERSIONS_DEFAULT_VALUE = false;
 
+    static final List<String> REMOVED_SETTINGS = List.of(
```
Comment (cody-littley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SettingConstants.java:106]: make this a set

**Review by cody-littley — APPROVED** [review]
This is a bit of a hack, but I'm willing to approve the PR since it solves a somewhat urgent problem.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Settings.java [hunk 2]
```diff
@@ -546,30 +547,33 @@ private boolean handleSetting(final String[] pars) {
...
-        final Field field = getFieldByName(Settings.class.getDeclaredFields(), name);
-        if (field != null && !Modifier.isFinal(field.getModifiers())) {
-            try {
-                if (subName == null) {
-                    good = setValue(field, this, val);
-                } else {
-                    final Field subField = getFieldByName(field.getType().getDeclaredFields(), subName);
-                    if (subField != null) {
-                        good = setValue(subField, field.get(this), val);
+        if (!REMOVED_SETTINGS.contains(name)) {
```
[the remark this answers is not in the document]
Comment (timo0) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Settings.java:550]: Done

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SettingConstants.java (hunk 1 above)
Comment (timo0) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SettingConstants.java]: Done

**Review by cody-littley — APPROVED** [review]

**Pull request merged**